process/sync/trieIterators/tokensSuppliesComputer.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/multiversx/mx-chain-core-go/data/esdt"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/common/errChan"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/dblookupext/esdtSupply"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/trie/keyBuilder"
+)
+
+type tokensSuppliesProcessor struct {
+	storageService dataRetriever.StorageService
+	marshaller     marshal.Marshalizer
+	tokensSupplies map[string]*big.Int
+}
+
+// ArgsTokensSuppliesProcessor is the arguments struct for NewTokensSuppliesProcessor
+type ArgsTokensSuppliesProcessor struct {
+	StorageService dataRetriever.StorageService
+	Marshaller     marshal.Marshalizer
+}
+
+// NewTokensSuppliesProcessor returns a new instance of tokensSuppliesProcessor
+func NewTokensSuppliesProcessor(args ArgsTokensSuppliesProcessor) (*tokensSuppliesProcessor, error) {
+	if check.IfNil(args.StorageService) {
+		return nil, errNilStorageService
+	}
+	if check.IfNil(args.Marshaller) {
+		return nil, errNilMarshaller
+	}
+
+	return &tokensSuppliesProcessor{
+		storageService: args.StorageService,
+		marshaller:     args.Marshaller,
+		tokensSupplies: make(map[string]*big.Int),
+	}, nil
+}
+
+// HandleTrieAccountIteration is the handler for the trie account iteration
+func (t *tokensSuppliesProcessor) HandleTrieAccountIteration(userAccount state.UserAccountHandler) error {
+	if check.IfNil(userAccount) {
+		return errNilUserAccount
+	}
+	if bytes.Equal(core.SystemAccountAddress, userAccount.AddressBytes()) {
+		log.Debug("repopulate tokens supplies: skipping system account address")
+		return nil
+	}
+	rh := userAccount.GetRootHash()
+	if len(rh) == 0 {
+		return nil
+	}
+
+	dataTrie := &common.TrieIteratorChannels{
+		LeavesChan: make(chan core.KeyValueHolder, common.TrieLeavesChannelDefaultCapacity),
+		ErrChan:    errChan.NewErrChanWrapper(),
+	}
+
+	errDataTrieGet := userAccount.DataTrie().GetAllLeavesOnChannel(dataTrie, context.Background(), rh, keyBuilder.NewKeyBuilder())
+	if errDataTrieGet != nil {
+		return errDataTrieGet
+	}
+
+	log.Trace("extractTokensSupplies - parsing account", "address", userAccount.AddressBytes())
+	esdtPrefix := []byte(core.ProtectedKeyPrefix + core.ESDTKeyIdentifier)
+	for userLeaf := range dataTrie.LeavesChan {
+		if !bytes.HasPrefix(userLeaf.Key(), esdtPrefix) {
+			continue
+		}
+
+		tokenKey := userLeaf.Key()
+		lenESDTPrefix := len(esdtPrefix)
+		suffix := append(userLeaf.Key(), userAccount.AddressBytes()...)
+		value, errVal := userLeaf.ValueWithoutSuffix(suffix)
+		if errVal != nil {
+			log.Warn("cannot get value without suffix", "error", errVal, "key", userLeaf.Key())
+			return errVal
+		}
+		var esToken esdt.ESDigitalToken
+		err := t.marshaller.Unmarshal(&esToken, value)
+		if err != nil {
+			return err
+		}
+
+		tokenName := string(tokenKey)[lenESDTPrefix:]
```

Comment:
no need, since we have this at the beginning of the `for` loop:
```
if !bytes.HasPrefix(userLeaf.Key(), esdtPrefix) {
			continue
		}
```
and also this: `lenESDTPrefix := len(esdtPrefix)`

Reply:
ok